numba/parfor.py

```diff
@@ -2887,6 +3032,18 @@ def parfor_typeinfer(parfor, typeinferer):
 
 typeinfer.typeinfer_extensions[Parfor] = parfor_typeinfer
 
+def build_parfor_definitions(parfor, definitions=None):
+    """get variable definition table for parfors"""
+    if definitions is None:
+        definitions = dict()
+
+    blocks = wrap_parfor_blocks(parfor)
```

Comment:
This wrap/unwrap pattern should also be a "with" style probably.

Reply:
I agree. There are subtle differences in its usage so it needs more major refactoring. Let's leave it for another PR.